### PR TITLE
Validate that Flyte labels are K8s compliant

### PR DESCRIPTION
**Pull request by kosigz-lyft**
## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/551

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/manager/impl/validation/validation.go [hunk 1]
```diff
@@ -196,6 +203,11 @@ func validateParameterMap(inputMap *core.ParameterMap, fieldName string) error {
 	return nil
 }
 
+func ValidateProjectUpdateRequestLabelsAlphanumeric(request admin.Project) error {
```
Comment (katrogan) [inline pkg/manager/impl/validation/validation.go:206]: can you call this in the ValidateProjectRegisterRequest (create) call too? :D

File: pkg/manager/impl/validation/validation.go [hunk 2]
```diff
@@ -196,6 +203,11 @@ func validateParameterMap(inputMap *core.ParameterMap, fieldName string) error {
...
+	// TODO(kosigz): implement this
```
Comment (katrogan) [inline pkg/manager/impl/validation/validation.go:207]: `validation.IsDNS1123Label()` should be sufficient!

File: pkg/manager/impl/validation/validation.go (hunk 1 above)
Comment (kosigz-lyft) [reply pkg/manager/impl/validation/validation.go:206]: Good call!

File: pkg/manager/impl/validation/validation.go (hunk 2 above)
Comment (kosigz-lyft) [reply pkg/manager/impl/validation/validation.go:207]: Thanks! 👍 On it.

File: pkg/manager/impl/validation/project_validator.go [hunk 3]
```diff
@@ -24,6 +24,12 @@ func ValidateProjectRegisterRequest(request admin.ProjectRegisterRequest) error
 	if err := ValidateEmptyStringField(request.Project.Id, projectID); err != nil {
 		return err
 	}
+	// Note: a bit confusingly named; the project that is registered should be the same as a project update
```
Comment (katrogan) [inline pkg/manager/impl/validation/project_validator.go:27]: can you just call ValidateProjectUpdateRequestLabelsAlphanumeric directly on the labels so we can avoid the confusing name? (and maybe call it ValidateLabelsAlphanumeric or something not referencing Update too :-) )

Comment (kosigz-lyft) [reply pkg/manager/impl/validation/project_validator.go:27]: Yeah, that seems like a good idea. Let me get through wiring up unit/integration tests with the current revision and then I'll modify the signatures.

Comment (katrogan) [reply pkg/manager/impl/validation/project_validator.go:27]: sg, feel free to ping when you're ready for a review

**Comment by kosigz-lyft** [comment]
PTAL #flyte-eng @katrogan

**Review by katrogan — APPROVED** [review]

**Pull request merged**